packages/sanity/src/presentation/loader/LiveQueries.tsx

```diff
@@ -55,55 +41,18 @@ export interface LoaderQueriesProps {
   ) => void
 }
 
-export default function LoaderQueries(props: LoaderQueriesProps): React.JSX.Element {
-  const {
-    liveDocument: _liveDocument,
-    controller,
-    perspective: activePerspective,
-    onLoadersConnection,
-    onDocumentsOnPage,
-  } = props
+export default function LiveQueries(props: LiveQueriesProps): React.JSX.Element {
+  const {controller, perspective: activePerspective, onLoadersConnection, onDocumentsOnPage} = props
 
   const [comlink, setComlink] = useState<ChannelInstance<LoaderControllerMsg, LoaderNodeMsg>>()
-  const [liveQueries, setLiveQueries] = useState<LiveQueriesState>({})
+  const [liveQueries, liveQueriesDispatch] = useLiveQueries()
 
   const projectId = useProjectId()
   const dataset = useDataset()
 
-  useEffect(() => {
-    const interval = setInterval(
-      () =>
-        setLiveQueries((liveQueries) => {
-          if (Object.keys(liveQueries).length < 1) {
-            return liveQueries
-          }
-
-          const now = Date.now()
-          const hasAnyExpired = Object.values(liveQueries).some(
-            // eslint-disable-next-line max-nested-callbacks
-            (liveQuery) =>
-              liveQuery.heartbeat !== false && now > liveQuery.receivedAt + liveQuery.heartbeat,
-          )
-          if (!hasAnyExpired) {
-            return liveQueries
-          }
-          const next = {} as LiveQueriesState
-          for (const [key, value] of Object.entries(liveQueries)) {
-            if (value.heartbeat !== false && now > value.receivedAt + value.heartbeat) {
-              continue
-            }
-            next[key] = value
-          }
-          return next
-        }),
-      MIN_LOADER_QUERY_LISTEN_HEARTBEAT_INTERVAL,
-    )
-    return () => clearInterval(interval)
-  }, [])
-
-  useEffect(() => {
+  useEffect((): (() => void) => {
```

Comment:
This typing:

```ts
useEffect((): (() => void) => {
           // ^
})
```
Is there to avoid easy mistakes, as further down this is returned:

```ts
return nextComlink.start()
```
Calling `.start()` returns an unsubscribe handler. This is a bit implicit and easy to miss. Should someone refactor this useEffect later and miss the `return` statement then TS will catch it 😮‍💨